===== Patch instructions generated by SAVER =====
[+] { Insert tmp := grub_usb_bulk_finish_readwrite:transfer->data_chunk at 5 (line 313, column 3) and replace the original expression at 3 (line 315, column 3) under true }
===== Actual patch in diff format =====
@@ -309,10 +309,11 @@ grub_usb_bulk_finish_readwrite (grub_usb_transfer_t transfer)
     grub_memcpy (transfer->data, (void *)
 		 grub_dma_get_virt (transfer->data_chunk),
 		 transfer->size + 1);
   grub_free (transfer->transactions);
+  tmp = transfer->data_chunk;
   grub_free (transfer);
-  grub_dma_free (transfer->data_chunk);
+  grub_dma_free (tmp);
 }
 
 static grub_usb_err_t


